src/main/java/com/example/hashcartapp/controller/AdvertisementController.java

```diff
@@ -0,0 +1,181 @@
+package com.example.hashcartapp.controller;
+
+import com.example.hashcartapp.dto.AdvertisementDTO;
+import com.example.hashcartapp.entities.Advertisement;
+import com.example.hashcartapp.service.AdvertisementService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.client.HttpClientErrorException;
+
+import javax.validation.Valid;
+import java.util.List;
+
+@RestController
+public class AdvertisementController {
+    @Autowired
+    private AdvertisementService advertisementService;
+
+	Logger logger = LoggerFactory.getLogger(AdvertisementController.class);
+
+	/**
+	 * It shows all the advertisements
+	 * @return It returns details of all the advertisements
+	 */
+	@GetMapping("/advertisement")
+    public List<AdvertisementDTO> getAdvertisements() {
+        return advertisementService.getAllAdvertisements();
+    }
+
+	/**
+	 * It shows the detail of the advertisement of any particular id
+	 * @param advertisementId
+	 * @return  It returns detail of the advertisement
+	 */
+	@GetMapping("/advertisement/{advertisementId}")
+    public ResponseEntity<AdvertisementDTO> getAdvertisementById(@PathVariable Long advertisementId) {
+		AdvertisementDTO advertisement = advertisementService.getAdvertisementById(advertisementId);
+		if (advertisement != null) {
+			return new ResponseEntity<AdvertisementDTO>(advertisement, HttpStatus.OK);
+		}
+		return new ResponseEntity<AdvertisementDTO>(HttpStatus.BAD_REQUEST);
+	}
+
+
+	/**
+	 * Show advertisement by category
+	 * @param page pageno
+	 * @param size size
+	 * @return Advertisement by category
+	 */
+	@GetMapping("/advertisementByCategory/{category}")
+	public Page<Advertisement> findAdvertisementByCategory(@PathVariable("category") String category,
+														   @RequestParam(value = "page", defaultValue = "1") Integer page,
+														   @RequestParam(value = "size", defaultValue = "3") Integer size) {
+		PageRequest request = PageRequest.of(page - 1, size);
+		return advertisementService.findAdvertisementByCategory(category,request);
+	}
+
+
+	/**
+	 *  Show advertisement by type
+	 */
+	@GetMapping("/advertisementByType/{type}")
+	public Page<Advertisement> findAdvertisementByType(@PathVariable("type") String type,
+													   @RequestParam(value = "page", defaultValue = "1") Integer page,
```

Comment:
if there are only a predefined number of types and categories, better to use enums